### PR TITLE
Endpoint which groups link promotions by container

### PR DESCRIPTION
**Pull request by ghost**
Some fixes - until I replace this App with one inside the Ophan project.
- /history.json shows all the positions a link has appeared in a container
- /history-by-container.json shows the cumulative amount of time a link stayed in a container
- Some refactoring to make the two endpoints share the same rendering code
- Position.index is now an Option[Int] because it is meaningless when viewing /history-by-container.json which takes no notice of link positions

### CODE REVIEW

File: src/main/scala/mercury/PageScanner.scala [hunk 1]
```diff
@@ -34,11 +34,17 @@ object PageScanner {
...
-      val isSublink = link.parents().asScala.exists(e => e.attr("data-link-name").contains("sublink"))
-      SimpleLink(href, comp, isSublink)
+    val links = elems.flatMap { link =>
+
+      val component = findDataComponent(link).filterNot {
+        c => c == "most-popular" || c.startsWith("popular-in-")
+      }
+
+      component.map { c =>
+        val href = link.attr("abs:href").takeWhile('?' != _).takeWhile('#' != _)
```
Comment (philwills) [inline src/main/scala/mercury/PageScanner.scala:44]: It looks like you're writing half a URL parser.  Have you considered https://github.com/smola/galimatias, which we use in Ophan?

**Comment by tackley** [comment]
+1

**Comment by philwills** [comment]
:+1:

File: src/main/scala/mercury/PageScanner.scala (hunk 1 above)
Comment (ghost) [inline src/main/scala/mercury/PageScanner.scala:44]: Agreed. This is really just a copy & paste job from what was already in Mercury. I'll look in to this for the the new version of Mercury (code named Freddie for now).

**Pull request merged**